tests/devices/test_default_qutrit.py

```diff
@@ -797,15 +830,171 @@ def test_hermitian_two_wires_identity_expectation(self, tol):
         expected = 3.5 * 1 * 1
         assert np.allclose(res, expected, atol=tol, rtol=0)
 
+    @pytest.mark.parametrize(
+        "index_1, eval_1", list(zip(list(range(1, 9)), [0, 0, -1, 0, 0, 0, 0, 1 / math.sqrt(3)]))
+    )
+    @pytest.mark.parametrize(
+        "index_2, eval_2", list(zip(list(range(1, 9)), [0, 0, 0, 0, 0, 0, 0, -2 / math.sqrt(3)]))
+    )
+    def test_gell_mann_tensor(self, index_1, index_2, eval_1, eval_2, tol):
+        """Test that the expectation value of the tensor product of two Gell-Mann observables is
+        correct"""
+        dev = qml.device("default.qutrit", wires=2)
+        obs = qml.GellMannObs(index_1, wires=0) @ qml.GellMannObs(index_2, wires=1)
+
+        dev.apply(
+            [
+                qml.QutritUnitary(TSHIFT, wires=0),
+                qml.QutritUnitary(TSHIFT, wires=1),
+                qml.QutritUnitary(TADD, wires=[0, 1]),
+            ],
+            obs.diagonalizing_gates(),
+        )
+        res = dev.expval(obs)
+
+        expected = eval_1 * eval_2
```

Comment:
Will this be true in general? Aren't there states for which the expval of a tensor product is not necessarily the product of expvals?

Reply:
I think that this test passes due to the fact that the measured state doesn't have any entanglement. I'll try out a circuit with entanglement and test whether this still works, which it likely wont, in which case I'll edit the expected result to work with the circuit.